tests/e2e/framework/kubernetes.go

```diff
@@ -0,0 +1,230 @@
+	}
+	usr, eUser := user.Current()
+	if eUser != nil {
+		return eUser
+	}
+	homeDir := usr.HomeDir
+
+	// Download Istioctl binary
+	util.WebDownload(k.TmpDir+"/istioctl", k.IstioCtlUrl+"/istioctl-linux")
+	if err := os.Chmod(fmt.Sprintf("%s/istioctl", k.TmpDir), 0755); err != nil {
+		return err
+	}
+	k.IstioCtl = fmt.Sprintf("%s/istioctl -c %s/.kube/config", k.TmpDir, homeDir)
+	glog.Infof("Downloaded istioctl to %s/istioctl\n", k.TmpDir)
+
+	if err := k.deployCore("manager.yaml"); err != nil {
+		return err
+	}
+	if err := k.deployCore("mixer.yaml"); err != nil {
+		return err
+	}
+	if err := k.deployCore("ingress-proxy.yaml"); err != nil {
+		return err
+	}
+	if err := k.deployCore("egress-proxy.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Deploy istio module from yaml files
+func (k *KubeInfo) deployCore(name string) error {
+	yamlFile := k.TmpDir + "/yaml/" + name
+	if err := util.Fill(yamlFile, name+".tmpl", *k); err != nil {
+		glog.Errorf("Failed to fill %s", yamlFile)
+		return err
+	}
+	if err := util.KubeApply(k.Namespace, yamlFile); err != nil {
+		glog.Errorf("Kubectl apply %s failed", yamlFile)
+		return err
+	}
+
+	return nil
+}
+
+// Deploy testing app
+func DeployApp(k *KubeInfo, deployment, svcName, port1, port2, port3, port4, version string, injectProxy bool) error {
+	yamlFile := k.TmpDir + "/yaml/" + svcName + "-app.yaml"
+	if err := util.Fill(yamlFile, "app.yaml.tmpl", map[string]string{
+		"Hub":        k.Hub,
+		"Tag":        k.Tag,
+		"service":    svcName,
+		"deployment": deployment,
+		"port1":      port1,
+		"port2":      port2,
+		"port3":      port3,
+		"port4":      port4,
+		"version":    version,
+	}); err != nil {
+		glog.Errorf("Failed to generate yaml for service %s in deployment %s", svcName, deployment)
+		return err
+	}
+
+	if injectProxy {
+		injectedYamlFile := k.TmpDir + "/yaml/injected-" + svcName + "-app.yaml"
+		if _, err := util.Shell(fmt.Sprintf("%s kube-inject -f %s -o %s --hub %s --tag %s -n %s", k.IstioCtl, yamlFile, injectedYamlFile, k.ProxyHub, k.ProxyTag, k.Namespace)); err != nil {
+			glog.Errorf("Kube-inject failed for service %s in deployment %s", svcName, deployment)
+			return err
+		}
+		yamlFile = injectedYamlFile
+	}
+
+	if err := util.KubeApply(k.Namespace, yamlFile); err != nil {
+		glog.Errorf("Kubectl apply %s failed", yamlFile)
+		return err
+	}
+	return nil
+}
+
+// Parse default value from istio.VERSION, TODO: find a better way to do this
+func parseVersion() {
+	file, err := os.Open(filepath.Join(util.GetTestRuntimePath(), VERSION_FILE))
+	if err != nil {
+		glog.Warningf("Cannot get version file %s : %s\n", VERSION_FILE, err)
+		return
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if m, _ := regexp.MatchString("^export (..*)=(..*)", line); m {
+			parts := strings.Split(line[7:], "=")
+			switch parts[0] {
+			case "MIXER_HUB":
+				mixerHubDefault = (parts[1])[1:(len(parts[1]) - 1)]
```

Comment:
Since you are iterating, it would be better to use a compiled regex:
r := regexp.MustCompile(`"^export (..*)=(..*)"`)
for scanner.Scan() {
 +		line := scanner.Text()
		if m := r.FindStringSubmatch(line); m != nil {
...
}